### DISCUSSION TITLE
Solve did not converge after a specific time step

### DISCUSSION THREAD

**Discussion by kingmantang** [idaholab/moose#20478] [Q&A Modules: General] [answered]
Hi, 

I am new to MOOSE and trying to apply it for study about thermal pressurization.
I am trying to study a 2D-quarter of borehole model with DirichletBC of Parsed function which should work with previous version.
However, convergence issue happens after a specific time step of around t = 0.013 with the update version. The error is listed below.

![moose fail](https://user-images.githubusercontent.com/100996887/156882915-51cad22a-d38c-4626-b7cb-b5272609d740.jpg)

I guess the converge fails when the material changes its state to plastic.
Are there any way to solve this error to visualize the change of temperature, porepressure and mechanical dissipation in the plastic state?
This is the [input file](https://github.com/kingmantang/Moose-TP) I am using. Thank you very much.

Best regards,
Kowin Tang

**Comment by GiudGiud** [comment]
@permcody who may we tag from the Redback team?
@fdkong these residuals are exactly 1e-10? This seems weird. Have you seen something like that before?
@kingmantang, with my very little knowledge about your simulation:
- I would first try to turn off the line search with `line_search = none` in the [Executioner] block. 
- Then I would add a whole bunch of debug output to try to get more information about what is happening. This block can be helpful:
```
[Debug]
  show_var_residual_norms = true
[]
```
- I would also compile in devel mode and see if any assertions are being hit. `METHOD=devel make` in the code repository then run the input file again

Guillaume

**Reply by hugary1995** [reply]
@vev001 is the right person to ask about redback questions.

**Comment by vev001** [comment, marked as answer]
Thanks @hugary1995 . We’ll take it from here with @pou036 . 

@kingmantang, we have a number of solver options and preconditioning in redback, depending on the problem. We’ll have a look at your input file and come back to you for the best solution.

**Comment by fdkong** [comment]
> these residuals are exactly 1e-10? This seems weird. Anything in petsc hard coded to output such a residual
 
Possibly not. We do not really hard code anything anywhere in PETSc. It seems to me that the residual is corresponding to some constant of equations when considering that the time step is super small.  The time step is so small that the right hand side of the linear system is zero, and then the direction is zero.  Noway to make any progress on the nonlinear level when the direction is zero. 

You might look into physics before you hit this step time. Check whether or not the physics still looks correct before you reach this time step.

**Reply by vev001** [reply]
We typically use these options in Redback to indeed tackle the stiff transition to plasticity when the full THMC system is solved (yes, it involves some convincing and often we hardcode PETSc haha). In this case there might be better solver options, the problem is not that stiff. I will check tomorrow and provide a suggested solution. Bear with me
